### PR TITLE
[Spoilers] Add new cog

### PR DESCRIPTION
**Pull request by Injabie3**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature

### Description of the changes
Add spoilers cog, addresses #105 .

#### Usage
`renspoiler <message>`

### CODE REVIEW

File: cogs/spoilers.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    #Class constructor
+    def __init__(self, bot):
+        self.bot = bot
+
+        #The JSON keys for the settings:
+        checkFolder()
+        checkFiles()
+        self.settings = config.Config("settings.json",
+                                      cogname="lui-cogs/spoilers")
```
Comment (KamiCreed) [inline cogs/spoilers.py:50]: I'm not sure, but does this create the `settings.json` file in the `spoilers` cog folder? No absolute paths needed?

File: cogs/spoilers.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        self.messages = self.settings.get("messages") if not None else {}
+
+    @commands.command(name="spoiler", pass_context=True)
+    async def spoiler(self, ctx, *, msg):
+        """Create a message spoiler.
+        Checks to see if the message contains the prefix, and if it does, it saves it
+        for later retrieval.
```
Comment (KamiCreed) [inline cogs/spoilers.py]: I guess this comment is out of date, since you made this into a command I think.

**Review by KamiCreed — APPROVED** [review]

**Comment by KamiCreed** [comment]
Hmm. Zach brings up a good point:

image

If someone dumb does that, it might almost DDOS the bot...

**Comment by Injabie3** [comment]
Perhaps add a 1 min cool down between 2 consecutive reacts per person? Thoughts on that?

**Comment by KamiCreed** [comment]
Probably could even be longer, since the DMed message doesn't really disappear, but I guess that's probably alright.

File: cogs/spoilers.py (hunk 1 above)
Comment (Injabie3) [reply cogs/spoilers.py:50]: It doesn't actually create the file, but it does fetch relative to where we start the bot, and defaults to the `data` folder.  It throws an error if the file doesn't exist, so we create the blank json in the `checkFiles()` function.

File: cogs/spoilers.py (hunk 2 above)
Comment (Injabie3) [reply cogs/spoilers.py]: Ah yes, fixed now.

**Comment by Injabie3** [comment]
Thanks for the feedback, let me know what you think of these new changes.

**Review by KamiCreed — APPROVED** [review]

**Pull request merged**